### DISCUSSION TITLE
HMIP-FROLL value LEVEL propagates wrong value

### DISCUSSION THREAD

**Discussion by dagobert** [OpenCCU/OpenCCU#2345] [🇺🇸 Q&A] [answered]
### Describe the issue you are experiencing

HMIP-FROLL does not propagate LEVEL when using UP or DOWN buttons in the UI after pushing stop before totally up or down. Instead when pushing the button down LEVEL will become value 0 and when pufhing the button up LEVEL will become 1 after stop button pressed while shutter was still moving.

When using the text input to eg. 30% LEVEL will become value 0.3 correctly.

The UI is always showing the correct shutter value but not the internal LEVEL value propagated to eg nodered.

### Describe the behavior you expected

LEVEL should propagate current level.

### Steps to reproduce the issue

1. shutter is open
2. press shutter button down
3. press shutter button stop while shutter is moving

In NodeRed I have created a simple flow catching the LEVEL value. 

### What is the version this bug report is based on?

3.69.7.20230506

### Which base platform are you running?

rpi3 (RaspberryPi3)

### Which HomeMatic/homematicIP radio module are you using?

RPI-RF-MOD

### Anything in the logs that might be useful for us?

```txt
I cannot exactly find out anymore when this bug started to appear. I used Version 3.65.11.20221005 before and it already had the problem. Now I am at 3.69.7.20230506
```

### Additional information

![Level Propagation Sequence](https://github.com/jens-maus/RaspberryMatic/assets/140021/5c451bf7-01a5-457f-aeea-c228f3529efa)

**Comment by dagobert** [comment, marked as answer]
I guess I found the problem:

The way LEVEL on Channel 4 is used has changed. Prior to some version of rasperrymatic (see above) on channel 4 LEVEL was propagating the value on the final state of the shutter. When you stopped the movement while eg. closing the shutter, LEVEL would become the value of the blinds's stopped position.
This changed, Now, when eg. closing the blinds and stopping them in motion LEVEL on channel 4 will be set to 0 which is not the actual position but more the command you gave to close them. Hence, I guess when setting a percent position and not stopping the movement the value of LEVEL on channel 4 is showing the actual position.

This new way was braking my automations but I guess this way it makes more sense. In expert mode one can see that channel 3 shows the actual position in percent. On channel 4 LEVEL represents the value one entered in the textbox and not the actual position.

Long story short: I might have missed this breaking change in the release notes. Furthermore it was probably a long existing bug no one really was aware of ore cared before.

**Reply by MichaelN0815** [reply]
Da gab es keinen breaking change. Kanal 4 war immer der Soll-Wert Kanal und Kanal 3 der IST-Wert Kanal. Evt. Hast Du jetzt erst vom vereinfachten Modus auf den Experten Modus umgeschaltet. Der vereinfachte Modus verbirgt das vor dem User.

**Reply by dagobert** [reply]
Nein, ich nutze Raspberrymatic mit NodeRed als Zusatzsoftware und den Channel 4 als Erkennung der Behanghohe schon lange. Das mit dem Expertenmodus hatte ich nur geschrieben, damit man die einzelnen Kanäle nachvollziehen kann.

Meine Beobachtungen rühren aus der Steuerung und dem Auslesen von dort. In der Raspberrymatic ui hat sich nichts verändert.

**Reply by MichaelN0815** [reply]
OK, dann ist es wohl ein Fehler in NodeRed

**Reply by dagobert** [reply]
Das ist unwahrscheinlich, denn NodeRed bekommt die Rohdaten direkt von Raspberrymatic unverändert.
Aber ok. Das jetzige Verhalten scheint korrekter zu sein.

**Comment by dagobert** [comment]
I really would love to get a short comment on this change if I missed it or if it was silently introduced. 
This bug report can probably be closed.

**Comment by jens-maus** [comment]
Thanks for having solved the issue yourself. I guess that this will be better suited in the discussion fora part of this repo, thus I will move it accordingly for later reference.
